sndio: 1.6.0 -> 1.7.0

S-NA opened the pull request:
###### Motivation for this change
Updates sndio to the latest version.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

S-NA committed:
sndio: 1.6.0 -> 1.7.0

Updates sndio to the latest version.

risicle commented:
Fails to build on macos (10.14):

```
mkdir -p /nix/store/mq3gi0vq1jarrzhk9qg8pn4pvhiyjnv5-sndio-1.7.0/bin /nix/store/mq3gi0vq1jarrzhk9qg8pn4pvhiyjnv5-sndio-1.7.0/share/man/man1
  cp midicat /nix/store/mq3gi0vq1jarrzhk9qg8pn4pvhiyjnv5-sndio-1.7.0/bin
  cp midicat.1 /nix/store/mq3gi0vq1jarrzhk9qg8pn4pvhiyjnv5-sndio-1.7.0/share/man/man1
  make[1]: Leaving directory '/private/tmp/nix-build-sndio-1.7.0.drv-0/sndio-1.7.0/midicat'
  post-installation fixup
  gzipping man pages under /nix/store/mq3gi0vq1jarrzhk9qg8pn4pvhiyjnv5-sndio-1.7.0/share/man/
  strip is /nix/store/gmhd52yac3w4zk8xqsgh97makjh8gi7l-cctools-binutils-darwin-927.0.2/bin/strip
  stripping (with command strip and flags -S) in /nix/store/mq3gi0vq1jarrzhk9qg8pn4pvhiyjnv5-sndio-1.7.0/lib  /nix/store/mq3gi0vq1jarrzhk9qg8pn4pvhiyjnv5-sndio-1.7.0/bin 
  patching script interpreter paths in /nix/store/mq3gi0vq1jarrzhk9qg8pn4pvhiyjnv5-sndio-1.7.0
  error: install_name_tool: can't open file: /nix/store/mq3gi0vq1jarrzhk9qg8pn4pvhiyjnv5-sndio-1.7.0/lib/libsndio.7.0.dylib (No such file or directory)
```

Rather than chasing the exact `.dylib` name in a `postFixup` all the time, it **does** seem to work if you simply add `fixDarwinDylibNames` to the build inputs if `stdenv.isDarwin`.

Also, obviously, could do with a rebase.

S-NA committed:
sndio: fix up for Darwin

Applying the suggestion by @risicle to use fixDarwinDylibNames instead
of manually chasing the exact .dylib name in postFixup.

risicle commented:
WFM, macos 10.14, non-nixos linux x86_64.
